### PR TITLE
replace C file stuff with fmt

### PR DESCRIPTION
**Pull request by neheb**
Simplifies the code and avoids static analyzer warnings.

Locally, the file seems to be created with 0600 permissions and because
of O_EXCL, it will fail if a file is present.

Signed-off-by: Rosen Penev <rosenp@gmail.com>

### CODE REVIEW

**Comment by neheb** [comment]
Maybe this should use ex somehow.

**Comment by neheb** [comment]
```
   /home/runner/work/gerbera/gerbera/src/main.cc:44:10: fatal error: fmt/os.h: No such file or directory
     44 | #include <fmt/os.h>
```

orly?

**Comment by neheb** [comment]
version minimum becomes 7. Unfortunate.

**Comment by whyman** [comment]
There is also `writeTextFile` in `tools`

File: src/main.cc [hunk 1]
```diff
@@ -327,48 +329,28 @@ int main(int argc, char** argv, char** envp)
...
 
-            // exit if the pidfile already exists
-            struct stat s;
-            errno = 0;
-            // for pidfile to be not there stat needs to fail and it needs
-            // to fail due to ENOENT (no such file or dir)
-            if (!(-1 == stat(pidfile->c_str(), &s) && errno == ENOENT)) {
+            try {
+                auto pidf = fmt::output_file(pidfile->c_str(), O_CREAT | O_EXCL | O_WRONLY);
+                // get the pid of our process
```
Comment (KarlStraussberger) [inline src/main.cc:334]: Add your comment here `O_EXCL, it will fail if a file is present`

**Comment by neheb** [comment]
Meh replaced with fopen instead. fmt 7.1 is a bit too new.

**Comment by neheb** [comment]
fopen creates a 0644 file though...

**Comment by neheb** [comment]
But so does open...

**Pull request closed without merging**